include/lexy/dsl/context_counter.hpp

```diff
@@ -206,6 +206,54 @@ struct _ctx_ceq<H, T...> : branch_base
 
 namespace lexyd
 {
+template <int Value>
+struct _equals
+{
+    template <typename T>
+    constexpr bool operator()(T id) const noexcept
+    {
+        return id == Value;
+    }
+};
+template <int Value>
+using equals = _equals<Value>;
```

Comment:
There is no need to provide convenience typedefs for these; we can just stick with `_ugly` names.